src/app/archive/archive_lib/processor.ml

```diff
@@ -524,12 +530,20 @@ module Block = struct
   let add_if_doesn't_exist (module Conn : CONNECTION) ~constraint_constants
       ({data= t; hash} : (External_transition.t, State_hash.t) With_hash.t) =
     let open Deferred.Result.Let_syntax in
-    match%bind find (module Conn) ~state_hash:hash with
+    match%bind find_opt (module Conn) ~state_hash:hash with
     | Some block_id ->
         return block_id
     | None ->
         let%bind parent_id =
-          find (module Conn) ~state_hash:(External_transition.parent_hash t)
+          if
+            External_transition.blockchain_length t <> Unsigned.UInt32.of_int 1
+          then
+            find (module Conn) ~state_hash:(External_transition.parent_hash t)
+          else
+            Conn.find
+              (Caqti_request.find Caqti_type.unit Caqti_type.int
+                 "SELECT count(id) + 1 FROM blocks")
```

Comment:
why is this chosen here?

Reply:
Wouldn't this make the block disconnected temporarily? And then the database may still use the parent-id

Reply:
Currently this is a hack for the genesis_block, we would only hit this case if the block_height = 1.